src/app/main.cpp

```diff
@@ -123,6 +124,14 @@ void showSplashScreen();
 void adjustFileDescriptorLimit();
 #endif
 
+enum class DefaultPreferencesMode
+{
+    Legacy,
+    Current
+};
+
+void handleChangedDefaults(DefaultPreferencesMode mode);
```

Comment:
maybe this function and `DefaultPreferencesMode` declaration should reside in upgrade.cpp?

Reply:
Have no objection.